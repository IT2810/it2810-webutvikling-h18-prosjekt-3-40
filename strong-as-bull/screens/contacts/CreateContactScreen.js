import React from 'react';

import {
    View,
    ScrollView,
    Text,
    TextInput,
    StyleSheet,
    TouchableOpacity,
} from 'react-native';

//Usable if choose photo:
//import { ImagePicker, Permissions} from 'expo';

import { Thumbnail} from "native-base";

export default class CreateContactScreen extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            first_name: '',
            last_name: '',
            email: 'JohnApple@america.com',
            phone_number: '99912345',
            company: 'Company',
            image: 'https://upload.wikimedia.org/wikipedia/commons/7/7c/Profile_avatar_placeholder_large.png',
            contact: null,
            key: null,
            chosenOne: null,
        };

    }

    componentDidMount() {
        try {
            this.setState({
                contact: this.props.navigation.state.params.contact,
                first_name: this.props.navigation.state.params.contact.first_name,
                last_name: this.props.navigation.state.params.contact.last_name,
                email: this.props.navigation.state.params.contact.email,
                phone_number: this.props.navigation.state.params.contact.phone_number,
                image: this.props.navigation.state.params.contact.image,
                company: this.props.navigation.state.params.contact.company,
                key: this.props.navigation.state.params.key,

            });

        } catch (error) {
            this.setState({contact: null})
        }
    }

    static navigationOptions = ({ navigation }) => {
        return {
            headerTitle: 'Contact Info'
        };
    };

    triggerAddContact(){
        this.props.navigation.state.params.addMethod(this.state.chosenOne);
    }

    triggerEditContact(){
        this.props.navigation.state.params.editMethod(this.state.chosenOne, this.state.key);
    }

    render()
    {
        let { image } = this.state;

        return (
            <ScrollView style={styles.container}>

                <View style={styles.picture}>
                    <Thumbnail source={{ uri: this.state.image}} large style={styles.thumbnail} />
                </View>

                <TextInput
                    style={styles.infoInput}
                    placeholder={
                        this.state.contact === null?
                            'First Name' : this.state.contact.first_name
                    }
                    onChangeText={(input) => {
                        this.setState({
                            first_name: input,
                        })
                    }}
                />
                <TextInput
                    style={styles.infoInput}
                    placeholder={
                        this.state.contact === null?
                            'Last Name' : this.state.contact.last_name
                    }
                    onChangeText={(input) => {
                        this.setState({
                            last_name: input,
                        })
                    }}
                />
                <TextInput
                    style={styles.infoInput}
                    placeholder={
                        this.state.contact === null?
                            'Phone number' : this.state.contact.phone_number
                    }
                    onChangeText={(input) => {
                        this.setState({
                            phone_number: input,
                        })
                    }}
                />
                <TextInput
                    style={styles.infoInput}
                    placeholder={
                        this.state.contact === null?
                            'E-mail' : this.state.contact.email
                    }
                    onChangeText={(input) => {
                        this.setState({
                            email: input,
                        })
                    }}
                />
                <TextInput
                    style={styles.infoInput}
                    placeholder={
                            'Company'
                    }
                    onChangeText={(input) => {
                        this.setState({
                            company: input,
                        })
                    }}
                />

                <TouchableOpacity style = {styles.addButton} onPress={() => {
                    if (this.state.first_name === '' || this.state.last_name === '') {
                        alert('Please enter at least first and last name')
                    } else {
                        let newCon = {
                            first_name: this.state.first_name,
                            last_name: this.state.last_name,
                            email: this.state.email,
                            phone_number: this.state.phone_number,
                            company: this.state.company,
                            image: this.state.image,
                        };
                        this.state.chosenOne = newCon;

                        if (this.state.contact === null) {
                            this.triggerAddContact();
                        } else {
                            this.triggerEditContact();
                        }

                        this.props.navigation.navigate('ContactList');

                    }
                }
                }>
                    <Text style={styles.buttonText}>+</Text>
                </TouchableOpacity>
            </ScrollView>
        )
    }}


const styles = StyleSheet.create({
    container: {
        backgroundColor: '#fff',
        flex: 1,
        paddingTop: 10,
    },
    title: {
        fontSize: 35,
        justifyContent: 'center',
        textAlign: 'center',
        paddingBottom: 30,
    },
    infoInput: {
        fontSize: 20,
        padding: 20,
        backgroundColor: "#eee",
    },

    picture: {
        alignItems: 'center',
        padding: 20,
    },
    thumbnail: {
        height: 200,
        width: 200,
        borderWidth: 5,
        borderColor: '#999',
    },
    addButton: {
    position: 'absolute',
        zIndex: 11,
        top: 20,
        right: 20,
        backgroundColor: '#ccc',
        width: 50,
        height: 50,
        borderRadius: 35,
        alignItems: 'center',
        justifyContent: 'center',
        elevation: 8
    },
    buttonText: {

    }
});