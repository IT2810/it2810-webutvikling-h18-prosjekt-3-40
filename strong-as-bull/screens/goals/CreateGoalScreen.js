import React from 'react';
import { StyleSheet, Text, TouchableOpacity, View, TextInput } from "react-native";


export default class CreateGoalScreen extends React.Component {
    static navigationOptions = {
        title: 'CreateGoal',
    };

    constructor(props) {
        super(props);
        this.state = {
            new_step_goal: 10000,
        };
    };

    setNewGoal() {
        this.setState({new_step_goal: Number(this.state.new_step_goal)});
        console.log(typeof this.state.new_step_goal);
        console.log(this.state.new_step_goal);
        if (typeof this.state.new_step_goal === "number") {
            const {params} = this.props.navigation.state;
            params.setGoal(this.state.new_step_goal);
            this.props.navigation.navigate('Goals');
        }
        else{
            alert('New step goal must be a number!');
        }
    }

    render() {
        return (
            <View style={styles.container}>

                <Text>Number of Steps:</Text>
                <TextInput
                    style={styles.textInput}
                    placeholder='10000'
                    onChangeText={(new_step_goal) => this.setState({new_step_goal: Number(new_step_goal)})}
                    placeholderTextColor='black'
                    underlineColorAndroid='transparent'/>
                <TouchableOpacity style={styles.addButton} onPress={() => this.setNewGoal()}>
                    <Text style={styles.addButtonText}>Confirm</Text>
                </TouchableOpacity>
            </View>

        );
    }
}
const styles = StyleSheet.create({
    container: {
        flex: 1,

        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center',
        padding: 20

    },
    textInput: {
        alignSelf: 'stretch',
        color: '#000',
        padding: 5,
        backgroundColor: '#cccccc',
        borderWidth: 1,
        borderColor: '#ededed',
        marginBottom: 10,
    },
    addButton: {
        position: 'absolute',
        backgroundColor: '#E91E63',
        width: 300,
        height: 50,
        borderRadius: 10,
        alignItems: 'center',
        justifyContent: 'center',
        elevation: 8,
        bottom: 100,
    },
    addButtonText: {
        color: '#fff',
        fontSize: 24,
    },

});
