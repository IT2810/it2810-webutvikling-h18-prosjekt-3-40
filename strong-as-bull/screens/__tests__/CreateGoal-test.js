import 'react-native';
import React from 'react';
import CreateGoalScreen from '../goals/CreateGoalScreen.js';
import renderer from 'react-test-renderer';

import CreateContactScreen from "../contacts/CreateContactScreen";


it('CreateContact snapshot', () => {
    const snap = renderer.create(
        <CreateGoalScreen/>
    ).toJSON();

    expect(snap).toMatchSnapshot();

});



it('test constructor and componentDidMount', () => {
    let instance = renderer.create(
        <CreateContactScreen/>
    ).getInstance();

    let testState = {
        first_name: '',
        last_name: '',
        email: 'JohnApple@america.com',
        phone_umber: '99912345',
        company: 'Company',
        image: 'https://upload.wikimedia.org/wikipedia/commons/7/7c/Profile_avatar_placeholder_large.png',
        contact: null,
        key: null,
        chosenOne: null
    };

});

