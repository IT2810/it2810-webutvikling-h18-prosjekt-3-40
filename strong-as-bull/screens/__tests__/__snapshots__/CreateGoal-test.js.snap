// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateContact snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Number of Steps:
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="10000"
    placeholderTextColor="black"
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#cccccc",
        "borderColor": "#ededed",
        "borderWidth": 1,
        "color": "#000",
        "marginBottom": 10,
        "padding": 5,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#E91E63",
        "borderRadius": 10,
        "bottom": 100,
        "elevation": 8,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "width": 300,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontSize": 24,
        }
      }
    >
      Confirm
    </Text>
  </View>
</View>
`;
